gpjax/decision_making/acquisition_functions/thompson_sampling.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2023 The JaxGaussianProcesses Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from dataclasses import dataclass
+from typing import Mapping
+
+from gpjax.dataset import Dataset
+from gpjax.decision_making.acquisition_functions.base import (
+    AbstractAcquisitionFunctionBuilder,
+    AcquisitionFunction,
+)
+from gpjax.decision_making.utils import OBJECTIVE
+from gpjax.gps import ConjugatePosterior
+from gpjax.typing import KeyArray
+
+
+@dataclass
+class ThompsonSampling(AbstractAcquisitionFunctionBuilder):
+    """
+    Form an acquisition function by drawing an approximate sample from the posterior,
+    using decoupled sampling as introduced in [Wilson et. al. (2020)](https://arxiv.org/abs/2002.09309).
+    """
+
+    num_rff_features: int = 500
```

Comment:
Thats a v large default. I've had a lot of success with just 10. But perhaps thats a bit too low. Any thoughts @thomaspinder ?

Reply:
I think the default in Trieste is 1000?

Reply:
Yeah 500 seems very high. Usually use 50-100 whenever I use RFFs.

Reply:
Ok, have changed to 100. This is also consistent with the default provided in the `sample_approx` method.